core/trino-main/src/test/java/io/trino/execution/TestFastForwardBranchTask.java

```diff
@@ -0,0 +1,111 @@
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), FAIL);
+        assertBranches(tableName, "main");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(asQualifiedName(tableName), "main", "not_found")))
+                .hasErrorCode(BRANCH_NOT_FOUND)
+                .hasMessage("line 1:1: Branch 'not_found' does not exist");
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(asQualifiedName(tableName), "not_found", "main")))
+                .hasErrorCode(BRANCH_NOT_FOUND)
+                .hasMessage("line 1:1: Branch 'not_found' does not exist");
+        assertBranches(tableName, "main");
+    }
+
+    @Test
+    void testFastForwardSameBranch()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), FAIL);
+        assertBranches(tableName, "main");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(asQualifiedName(tableName), "main", "main")))
+                .hasErrorCode(GENERIC_USER_ERROR)
+                .hasMessage("line 1:1: Fast forwarding branch between same branches is not supported");
+        assertBranches(tableName, "main");
+    }
+
+    @Test
+    void testFastForwardBranchOnNotExistingTable()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(asQualifiedName(tableName), "main", "main")))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessage("line 1:1: Table 'test_catalog.schema.existing_table' does not exist");
+    }
+
+    @Test
+    void testFastForwardBranchOnView()
+    {
+        QualifiedObjectName viewName = qualifiedObjectName("existing_view");
+        metadata.createView(testSession, viewName, someView(), ImmutableMap.of(), false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(asQualifiedName(viewName), "main", "main")))
+                .hasErrorCode(GENERIC_USER_ERROR)
+                .hasMessage("line 1:1: Fast forwarding branch on view is not supported");
+    }
+
+    @Test
+    void testFastForwardBranchOnMaterializedView()
+    {
+        QualifiedName viewName = qualifiedName("existing_materialized_view");
+        metadata.createMaterializedView(testSession, QualifiedObjectName.valueOf(viewName.toString()), someMaterializedView(), MATERIALIZED_VIEW_PROPERTIES, false, false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeFastForwardBranch(viewName, "main", "main")))
+                .hasErrorCode(GENERIC_USER_ERROR)
+                .hasMessage("line 1:1: Fast forwarding branch on materialized view is not supported");
+    }
+
+    private ListenableFuture<Void> executeFastForwardBranch(QualifiedName tableName, String fromBranch, String toBranch)
```

Comment:
Can you add test case with successful fast forward?

Reply:
I don't think we can write meaningful tests without connectors.

Reply:
How about just checking that `executeFastForwardBranch` does not fail?